client/lib/happychat/test/index.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { stub } from 'sinon';
+import { EventEmitter } from 'events';
+
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_CONNECTING,
+	HAPPYCHAT_CONNECTED,
+	HAPPYCHAT_DISCONNECTED,
+	HAPPYCHAT_RECEIVE_EVENT,
+	HAPPYCHAT_RECONNECTING,
+	HAPPYCHAT_SET_AVAILABLE,
+	HAPPYCHAT_SET_CHAT_STATUS,
+	HAPPYCHAT_TRANSCRIPT_REQUEST,
+} from 'state/action-types';
+
+import buildConnection from '../connection';
+
+describe( 'connection', ( ) => {
+	describe( 'should bind socket ', ( ) => {
+		const user = {
+			signer_user_id: 12,
+			jwt: 'jwt',
+			locale: 'locale',
+			groups: 'groups',
+			geo_location: 'location',
+		};
+
+		let openSocket, socket, dispatch;
+		beforeEach( () => {
+			socket = new EventEmitter();
+			dispatch = stub();
+			const connection = buildConnection();
+			openSocket = connection.init( socket, dispatch, user );
+		} );
+
+		it( 'connect event', ( done ) => {
+			openSocket.then( ( ) => {
+				// TODO: implement when connect event is used
+				expect( true ).to.equal( true );
+				done(); // tell mocha the promise chain ended
+			} );
+			socket.emit( 'connect' );
+			socket.emit( 'init' ); // force openSocket promise to resolve
+		} );
+
+		it( 'token event', ( done ) => {
+			const callback = stub();
+			openSocket.then( ( ) => {
+				expect( callback.getCall( 0 ).args[ 0 ].signer_user_id ).to.equal( user.signer_user_id );
+				expect( callback.getCall( 0 ).args[ 0 ].jwt ).to.equal( user.jwt );
+				expect( callback.getCall( 0 ).args[ 0 ].locale ).to.equal( user.locale );
+				expect( callback.getCall( 0 ).args[ 0 ].groups ).to.equal( user.groups );
+				done(); // tell mocha the promise chain ended
+			} );
+			socket.emit( 'token', callback );
+			socket.emit( 'init' ); // force openSocket promise to resolve
+		} );
+
+		it( 'init event', ( done ) => {
+			openSocket.then( ( ) => {
+				expect( dispatch.getCall( 0 ).args[ 0 ].type ).to.equal( HAPPYCHAT_CONNECTING );
+				expect( dispatch.getCall( 1 ).args[ 0 ].type ).to.equal( HAPPYCHAT_CONNECTED );
+				expect( dispatch.getCall( 1 ).args[ 0 ].user.signer_user_id ).to.equal( user.signer_user_id );
+				expect( dispatch.getCall( 1 ).args[ 0 ].user.locale ).to.equal( user.locale );
+				expect( dispatch.getCall( 1 ).args[ 0 ].user.groups ).to.equal( user.groups );
+				expect( dispatch.getCall( 1 ).args[ 0 ].user.geo_location ).to.equal( user.geo_location );
+				expect( dispatch.getCall( 2 ).args[ 0 ].type ).to.equal( HAPPYCHAT_TRANSCRIPT_REQUEST );
```

Comment:
We could make this test simpler by applying the similar changes I [noted above](https://github.com/Automattic/wp-calypso/pull/18519/files#r143234771)